src/VisualStudio/Core/Test.Next/Services/ServiceHubServicesTests.cs

```diff
@@ -358,18 +350,19 @@ public void TestRemoteWorkspaceCircularReferences()
                 });
 
             var languages = ImmutableHashSet.Create(LanguageNames.CSharp);
-            var remoteWorkspace = new RemoteWorkspace(workspaceKind: "test");
+
+            using var remoteWorkspace = new RemoteWorkspace(FeaturesTestCompositions.RemoteHost.GetHostServices(), WorkspaceKind.RemoteWorkspace);
             var optionService = remoteWorkspace.Services.GetRequiredService<IOptionService>();
             var options = new SerializableOptionSet(languages, optionService, ImmutableHashSet<IOption>.Empty, ImmutableDictionary<OptionKey, object>.Empty, ImmutableHashSet<OptionKey>.Empty);
 
             // this shouldn't throw exception
-            remoteWorkspace.TryAddSolutionIfPossible(solutionInfo, workspaceVersion: 1, options, out var solution);
+            remoteWorkspace.TrySetCurrentSolution(solutionInfo, workspaceVersion: 1, options, out var solution);
```

Comment:
just so i understand, what'st he 'try' portion of it.  Is it 'try' because teh network call coudl fail?  or because multiple people may be doing thsi and only one should win?  etc. etc.?

Reply:
It returns false if attempting to set older version than its already the current.

```C#
                if (workspaceVersion <= _currentRemoteWorkspaceVersion)
                {
                    // we never move workspace backward
                    solution = null;
                    return false;
                }
```